token/cli/src/main.rs

```diff
@@ -648,7 +648,16 @@ fn command_transfer(
     memo: Option<String>,
     bulk_signers: BulkSigners,
     no_wait: bool,
+    yolo: bool,
 ) -> CommandResult {
+    let recipient_account = config.rpc_client.get_account(&recipient).unwrap();
+    if recipient_account.owner != system_program::id() && !yolo {
+        return Err(
+            ("Recipient is not a Account owned by System Program. Use --yolo if you know what you doing").to_string()
+                .into(),
+        );
+    }
+
```

Comment:
this needs to be moved into the block at ln705 where `recipient_is_token_account` is resolved taking into account:
1. No RPC calls are allowed in sign-only mode.  Probably fine because the account is already queried in that block, so you can just reuse it.
2. The recipient may have been specified as a token account directly. In which case, it is owned by the SPL Token program and this check should not be performed at all.